lib/postgrex/replication.ex

```diff
@@ -352,6 +401,49 @@ defmodule Postgrex.Replication do
     end
   end
 
+  @doc false
+  def handle_call({:start_replication, _opts}, _from, %{replication_started: true} = s),
+    do: {:reply, {:error, :replication_started}, s}
+
+  @doc false
+  def handle_call({:start_replication, opts}, _from, s) do
+    %{protocol: protocol} = s
+    statement = command(:start_replication, opts)
+
+    with {:ok, protocol} <- Protocol.handle_replication(statement, protocol),
+         {:ok, protocol} <- Protocol.checkin(protocol) do
+      {:reply, :ok, %{s | protocol: protocol, replication_started: true}}
+    else
+      {:error, reason, protocol} ->
+        {:reply, {:error, reason}, %{s | protocol: protocol}}
+
+      {:disconnect, reason, protocol} ->
+        reconnect_or_stop(:disconnect, reason, protocol, s)
+    end
+  end
+
+  @doc false
+  def handle_call({name, _opts}, _from, %{replication_started: true} = s)
```

Comment:
I think you can move this up and remove the duplication with a similar clause above?